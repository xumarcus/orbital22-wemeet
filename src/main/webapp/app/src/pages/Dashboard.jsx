import Typography from '@mui/material/Typography'
import * as React from 'react'
import { useContext } from 'react'
import Tabs from '@mui/material/Tabs'
import Tab from '@mui/material/Tab'
import Box from '@mui/material/Box'
import PageTitle from '../components/core/PageTitle'
import EventGrid from '../components/dashboard/EventGrid'
import AppContext from '../core/AppContext'
import ErrorFallback from '../components/core/ErrorFallback'
import { ErrorBoundary } from 'react-error-boundary'
import PendingResponseGrid from '../components/dashboard/PendingResponseGrid'

const TabPanel = ({ children, value, index, ...other }) => (
  <div
    role='tabpanel'
    hidden={value !== index}
    id={`simple-tabpanel-${index}`}
    aria-labelledby={`simple-tab-${index}`}
    {...other}
  >
    {value === index && (
      <Box sx={{ p: 3 }}>
        <Typography>{children}</Typography>
      </Box>
    )}
  </div>
)

const a11yProps = (index) => (
  {
    id: `simple-tab-${index}`,
    key: index,
    'aria-controls': `simple-tabpanel-${index}`
  }
)

const TABS = [
  'Events',
  'Pending Response'
]

const Dashboard = () => {
  const { context } = useContext(AppContext)
  const [index, setIndex] = React.useState(0)

  return (
    <>
      <PageTitle pageTitle='Dashboard' />
      <Box sx={{
        width: '100%',
        alignContent: 'center',
        justifyContent: 'center'
      }}
      >
        <Tabs
          value={index} onChange={(event, newIndex) => setIndex(newIndex)}
        >
          {TABS.map((label, index) => <Tab label={label} {...a11yProps(index)} />)}
        </Tabs>

        <Box sx={{ backgroundColor: '#efefef', height: '100vh' }}>
          <TabPanel value={index} index={0}>
            <ErrorBoundary FallbackComponent={ErrorFallback} resetKeys={[context]}>
              <EventGrid />
            </ErrorBoundary>
          </TabPanel>
          <TabPanel value={index} index={1}>
            <ErrorBoundary FallbackComponent={ErrorFallback} resetKeys={[context]}>
              <PendingResponseGrid />
            </ErrorBoundary>
          </TabPanel>
        </Box>
      </Box>
    </>
  )
}

export default Dashboard
