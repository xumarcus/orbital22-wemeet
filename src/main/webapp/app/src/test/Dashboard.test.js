import React from 'react'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'
import Dashboard from '../pages/Dashboard'

describe('Dashboard', () => {
  it('should render correctly when not signed in', () => {
    const container = document.createElement('div')
    document.body.appendChild(container)

    act(() => {
      render(<Dashboard />, container)
    })

    expect(container.innerHTML).toMatchSnapshot()

    unmountComponentAtNode(container)
    container.remove()
  })
})


