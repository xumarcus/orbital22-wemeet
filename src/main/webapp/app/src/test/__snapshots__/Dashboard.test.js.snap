// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should render correctly 1`] = `"<div class=\\"MuiTypography-root MuiTypography-h3 css-ugw3x7-MuiTypography-root\\">Dashboard</div><div class=\\"MuiBox-root css-11n9b3d\\"><div class=\\"MuiTabs-root css-1ujnqem-MuiTabs-root\\"><div class=\\"MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller\\" style=\\"overflow: hidden; margin-bottom: 0px;\\"><div class=\\"MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer\\" role=\\"tablist\\"><button class=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root\\" tabindex=\\"0\\" type=\\"button\\" role=\\"tab\\" aria-selected=\\"true\\" id=\\"simple-tab-0\\" aria-controls=\\"simple-tabpanel-0\\">Events<span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root\\" tabindex=\\"-1\\" type=\\"button\\" role=\\"tab\\" aria-selected=\\"false\\" id=\\"simple-tab-1\\" aria-controls=\\"simple-tabpanel-1\\">Pending Response<span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div><span class=\\"MuiTabs-indicator css-1aquho2-MuiTabs-indicator\\" style=\\"left: 0px; width: 0px;\\"></span></div></div><div class=\\"MuiBox-root css-18gompf\\"><div role=\\"tabpanel\\" id=\\"simple-tabpanel-0\\" aria-labelledby=\\"simple-tab-0\\"><div class=\\"MuiBox-root css-19kzrtu\\"><p class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-l6hioc-MuiGrid-root\\" style=\\"min-height: 80vh;\\"><div class=\\"MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root\\"><h1 class=\\"MuiTypography-root MuiTypography-h1 MuiTypography-alignCenter css-sqoje6-MuiTypography-root\\">⚠</h1><h5 class=\\"MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-ph2xsj-MuiTypography-root\\">Please sign in.</h5></div></div></p></div></div><div role=\\"tabpanel\\" hidden=\\"\\" id=\\"simple-tabpanel-1\\" aria-labelledby=\\"simple-tab-1\\"></div></div></div>"`;

exports[`Dashboard should render correctly when not signed in 1`] = `"<div class=\\"MuiTypography-root MuiTypography-h3 css-ugw3x7-MuiTypography-root\\">Dashboard</div><div class=\\"MuiBox-root css-11n9b3d\\"><div class=\\"MuiTabs-root css-1ujnqem-MuiTabs-root\\"><div class=\\"MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller\\" style=\\"overflow: hidden; margin-bottom: 0px;\\"><div class=\\"MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer\\" role=\\"tablist\\"><button class=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root\\" tabindex=\\"0\\" type=\\"button\\" role=\\"tab\\" aria-selected=\\"true\\" id=\\"simple-tab-0\\" aria-controls=\\"simple-tabpanel-0\\">Events<span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root\\" tabindex=\\"-1\\" type=\\"button\\" role=\\"tab\\" aria-selected=\\"false\\" id=\\"simple-tab-1\\" aria-controls=\\"simple-tabpanel-1\\">Pending Response<span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div><span class=\\"MuiTabs-indicator css-1aquho2-MuiTabs-indicator\\" style=\\"left: 0px; width: 0px;\\"></span></div></div><div class=\\"MuiBox-root css-18gompf\\"><div role=\\"tabpanel\\" id=\\"simple-tabpanel-0\\" aria-labelledby=\\"simple-tab-0\\"><div class=\\"MuiBox-root css-19kzrtu\\"><p class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-l6hioc-MuiGrid-root\\" style=\\"min-height: 80vh;\\"><div class=\\"MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root\\"><h1 class=\\"MuiTypography-root MuiTypography-h1 MuiTypography-alignCenter css-sqoje6-MuiTypography-root\\">⚠</h1><h5 class=\\"MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-ph2xsj-MuiTypography-root\\">Please sign in.</h5></div></div></p></div></div><div role=\\"tabpanel\\" hidden=\\"\\" id=\\"simple-tabpanel-1\\" aria-labelledby=\\"simple-tab-1\\"></div></div></div>"`;
